### ISSUE TITLE
All requests return HTTP 404

### ISSUE THREAD

**Issue by jeremy-jameson** [technology-toolbox/website#119]
It appears there was recently a change to Azure Static Web Apps that broke the Technology Toolbox website.

![image](https://user-images.githubusercontent.com/1385288/165562007-56821466-4875-4b6b-8a23-884665cf0601.png)

The custom HTTP 404 error page is shown (albeit with no styling and broken images) regardless of the URL -- for example:

- https://www.technologytoolbox.com/
- https://www.technologytoolbox.com/index.html
- https://www.technologytoolbox.com/blog/
- https://www.technologytoolbox.com/blog/jjameson/
- https://www.technologytoolbox.com/blog/jjameson/2011/08/22/leaving-microsoft/

I originally suspected the problem was due to a change to Oryx (the build engine used for Azure Static Web Apps) -- specifically, I thought the content generated by Hugo was no longer being copied as expected to the Azure Static Web App.

However, further investigation revealed the issue was with the Azure Static Web App itself (which was created in February 2021 -- when Azure Static Web Apps was still in "preview").

While troubleshooting, I stumbled upon the following:

[Azure Static Web App returns 404](https://docs.microsoft.com/en-us/answers/questions/803074/azure-static-web-app-returns-404.html)

Apparently somebody else encountered the same issue on a different Azure Static Web App about three weeks ago.

Here is the [latest response from someone at Microsoft on that issue](https://docs.microsoft.com/en-us/answers/questions/803074/azure-static-web-app-returns-404.html?childToView=819290#comment-819290):

> **SnehaAgrawal-MSFT** · 7 days ago
> Update: Product group is aware of this issue and is investigating further.
>
> In case any one else experiencing the same issue, please reply here. Will keep posted here.

I attempted to reply to the issue on the Microsoft Q&A website -- but, sadly, clicking the **Submit** button does nothing (which is *incredibly aggravating*...and, yes, I attempted with multiple browsers -- specifically Chrome and Edge).

I then proceeded to troubleshoot what I thought to be an issue with the `staticwebapp.config.json` file. However, I discovered that a freshly created Azure Static Web App using a copy of [the `staticwebapp.config.json` file for the Technology Toolbox website](https://github.com/technology-toolbox/website/blob/main/static/staticwebapp.config.json) worked without issue.

Consequently, I chose to rebuild the Azure Static Web App for the Technology Toolbox website in order to resolve the errors without having to wait on Microsoft to identify the root cause and implement a fix.

Rebuilding an Azure Static Web App is relatively easy, but it does come with some caveats:

- The hostname for the website is changed to a new random value (e.g. **nice-wave-0f7d7c51e.azurestaticapps.net** --> **thankful-pond-09e2c5f10.1.azurestaticapps.net**)
- Corresponding DNS records must be updated (i.e. the **A** record for **technologytoolbox.com** and the **CNAME** record for **www.technologytoolbox.com**)
- A new GitHub workflow YAML file is automatically added to the repo during the process of creating the replacement Azure Static Web App (e.g. `azure-static-web-apps-thankful-pond-09e2c5f10.yml`)
- While there are two GitHub workflow YAML files present in the repository, the GitHub Action will deploy to two different Azure Static Web Apps (i.e. the old and the new)
- Previous customizations to the old GitHub workflow YAML file (e.g. `azure-static-web-apps-nice-wave-0f7d7c51e.yml`) need to be merged into the new GitHub workflow YAML (e.g. `azure-static-web-apps-thankful-pond-09e2c5f10.yml`)
- The old GitHub workflow YAML (e.g. `azure-static-web-apps-nice-wave-0f7d7c51e.yml`) needs to be deleted
- Any references to the old hostname for the website need to be replaced with the new hostname -- specifically, for Technology Toolbox, this involves updating the list of domain names allowed in Commento to support testing a temporary "staging" environment created for a pull request)

**Comment by jeremy-jameson** [comment]
After rebuilding the Azure Static Web App and making the corresponding infrastructure changes (e.g. updating DNS records), the website is functioning as expected.

**Comment by jeremy-jameson** [comment]
Microsoft apparently fixed the issue with the Q&A site. I posted a reply and an answer this morning:

[Azure Static Web App returns 404](https://docs.microsoft.com/en-us/answers/questions/803074/azure-static-web-app-returns-404.html)
